Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -209,9 +207,21 @@ namespace PhysX
         *m_shapeConfig = Utils::CreateHeightfieldShapeConfiguration(GetEntityId());
     }
 
-    void EditorHeightfieldColliderComponent::RefreshHeightfield(const AZ::Aabb& dirtyRegion,
-        [[maybe_unused]] const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask changeMask)
+    void EditorHeightfieldColliderComponent::RefreshHeightfield(
+        const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask changeMask,
+        const AZ::Aabb& dirtyRegion)
     {
+        using Physics::HeightfieldProviderNotifications;
+
+        if (HeightfieldChangeMask::DestroyBegin == (changeMask & HeightfieldChangeMask::DestroyBegin) ||
```

Comment:
Might be nice (and more robust) to have utility functions to check if flags are set, rather than having to do the logic yourself every time.